runtime/lua/vim/filetype/detect.lua

```diff
@@ -1256,17 +1256,15 @@ end
 -- 2. Check the first 1000 non-comment lines for LaTeX or ConTeXt keywords.
 -- 3. Default to "plain" or to g:tex_flavor, can be set in user's vimrc.
 function M.tex(path, bufnr)
-  local format = getlines(bufnr, 1):find('^%%&%s*(%a+)')
-  if format then
+  local matched, _, format = getlines(bufnr, 1):find('^%%&%s*(%a+)')
+  if matched then
     format = format:lower():gsub('pdf', '', 1)
-    if format == 'tex' then
-      return 'tex'
-    elseif format == 'plaintex' then
-      return 'plaintex'
-    end
   elseif path:lower():find('tex/context/.*/.*%.tex') then
     return 'context'
   else
+    -- Default value, may be changed later:
+    format = vim.g.tex_flavor and vim.g.tex_flavor or 'plaintex'
```

Comment:
Can be simplified to `format = vim.g.tex_flavor or 'plainex'`

Reply:
Indeed, done (force pushed)